components/scream/src/physics/p3/tests/p3_saturation_unit_tests.cpp

```diff
@@ -26,23 +26,76 @@ template <typename D>
 struct UnitWrap::UnitTest<D>::TestP3Saturation
 {
 
-  KOKKOS_FUNCTION  static void saturation_tests(const Scalar& temperature, const Scalar& pressure, const Scalar& correct_sat_ice_p,
-    const Scalar& correct_sat_liq_p, const Scalar&  correct_mix_ice_r, const Scalar& correct_mix_liq_r, int& errors ){
+  static Scalar condNum(const Scalar& svp, const Scalar& temp, const bool isIce=false){
+
+    //computes condition number for saturation vapor pressure calc.
+
+    //Set constant values
+    //--------------------------------------
+    const Scalar RV = C::RV;
+    const Scalar RhoLiq = C::RhoH2O;
+    const Scalar RhoIce = C::RhoIce;
+    const Scalar LatVap = C::LatVap;
+    const Scalar LatIce = C::LatIce;
+
+    //==========================================================
+    // Test Saturation Vapor Pressure
+    //======================================================5A5A5A====
+    // First calculate condition number Cond=x*f'(x)/f(x), which gives the growth in roundoff error
+    // expected from the function. Here, x=temperature, f(x) = sat_X_p, and
+    // f'(x) = L/T*sat_X_p/(Rv*T - sat_X_p/rho_{liq or ice}) from Clausius-Clapeyron.
+    // Note this analytical solution isn't exact because it misses things like surface tension
+    // so it can't replace curve fits like Flatau. But Clausius-Clapeyron is good enough for
+    // getting a ballpark value, which is all we're doing here.
+
+    //for liquid
+    auto latent = LatVap;
+    auto rho    = RhoLiq;
+
+    if(isIce){
+      //for ice
+      latent += LatIce;
+      rho     = RhoIce;
+    }
+
+    return latent*svp/(RV*temp - svp/rho); //return condition number
+
+    }
+
+  KOKKOS_FUNCTION  static void saturation_tests(const Scalar& temperature, const Scalar& pressure,
+						//Flatau (polysvp1) correct values
+						const Scalar& correct_sat_ice_fp, const Scalar& correct_sat_liq_fp,
+						const Scalar& correct_mix_ice_fr, const Scalar& correct_mix_liq_fr,
+						//Muphy and Koop (MurphyKoop_svp) correct values
+						const Scalar& correct_sat_ice_mkp, const Scalar& correct_sat_liq_mkp,
+						const Scalar& correct_mix_ice_mkr, const Scalar& correct_mix_liq_mkr,
+						int& errors ){
+    //Nomenclature:
+    //subscript "_fp"  stands for "Flatau Pressure"
+    //subscript "_fr"  stands for "Flatau mixing Ratios"
+    //subscript "_mkp" stands for "Murphy Koop Pressure"
+    //subscript "_mkr" stands for "Murphy Koop mixing Ratios"
 
     //Allow usage of saturation functions
     using physics = scream::physics::Functions<Scalar, Device>;
-    
+
     //Convert Scalar inputs to Spacks because that's what polysvp1 and qv_sat expect as inputs.
     //--------------------------------------
     const Spack temps(temperature);
     const Spack pres(pressure);
 
-    //Get values from polysvp1 and qv_sat to test against "correct" values
+    //Get values from polysvp1 and qv_sat (calling polysvp1) to test against "correct" values
```

Comment:
this comment isn't correct since it "gets values from polysvp1 or MurphyKoop_svp or qv_sat to test..."

Reply:
In these calls, first we call polysvp1 and then we call qv_sat, which calls polysvp1. That is what the above comment is trying to say.

In next set of calls (~lines 94 to 98), we call MurphyKoop_svp and then we call qv_sat, which calls MurphyKoop_svp. I can reword the comments to make it more clear.